examples/discovery/discovery.go

```diff
@@ -0,0 +1,61 @@
+// Copyright 2018-2020 opcua authors. All rights reserved.
+// Use of this source code is governed by a MIT-style license that can be
+// found in the LICENSE file.
+
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"github.com/gopcua/opcua"
+	"github.com/gopcua/opcua/debug"
+	"log"
+)
+
+func main() {
+	endpoint := flag.String("endpoint", "opc.tcp://localhost:4840", "OPC UA Endpoint URL")
+	flag.BoolVar(&debug.Enable, "debug", false, "enable debug logging")
+	flag.Parse()
+	log.SetFlags(0)
+
+	ctx := context.Background()
+
+	findServersOnNetwork(ctx, *endpoint)
```

Comment:
I'd make the functions return an error in both cases and `log.Print(err)` in the one case and `log.Fatal(err)` in the other.